tests/tools.py

```diff
@@ -163,6 +166,21 @@ def emulate_assertIsNotNone(self, x, msg=None):
             msg = msg if msg is not None else '{0} is None'.format(_safe_repr(x))
             self.fail(msg)
 
+    def emulate_raises_regex(self, exception_type, regex, function, *args, **kwargs):
+        if (PY_VERSION_MAJOR == 2 and PY_VERSION_MINOR >= 7) or (PY_VERSION_MAJOR == 3 and PY_VERSION_MINOR >= 2):
```

Comment:
Don't test for the version number here; test for the availability of `assertRaisesRegexp`, the thing you'd like to use. That is...
```
if hasattr(cls, "assertRaisesRegexp"):
    with self.assertRaisesRegexp(exception_type, regex):
        function(*args, **kwargs)
else:
    try:
        function(*args, **kwargs)
    except
etc
```

Reply:
Ahh, fixed it a little different above, but same thing.

---
In reply to: [207340686](https://github.com/Azure/WALinuxAgent/pull/1281#discussion_r207340686) [](ancestors = 207340686)